stdlib/src/commonMain/kotlin/it/unibo/collektive/stdlib/Gossip.kt

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2024, Danilo Pianini, Nicolas Farabegoli, Elisa Tronetti,
+ * and all authors listed in the `build.gradle.kts` and the generated `pom.xml` file.
+ *
+ * This file is part of Collektive, and is distributed under the terms of the Apache License 2.0,
+ * as described in the LICENSE file in this project's repository's top directory.
+ */
+
+package it.unibo.collektive.stdlib
+
+import it.unibo.collektive.aggregate.api.Aggregate
+import it.unibo.collektive.aggregate.api.operators.share
+import it.unibo.collektive.field.Field.Companion.fold
+
+/**
+ * Gossip algorithm implementation.
+ * Each node starts with an [initial] value, shares it with its neighbors,
+ * and updates its value based on the best value received according to the [selector]. uik
+ */
+fun <ID : Comparable<ID>, Value> Aggregate<ID>.gossip(
+    initial: Value,
+    selector: Comparator<Value>,
+): Value {
+    val local = GossipValue<ID, Value>(initial, initial)
+    return share(local) { gossip ->
+        val result = gossip.fold(local) { current, next ->
+            val actualNext = if (localId in next.path) next.base() else next
+            val comparison = selector.compare(current.best, actualNext.best)
+            when {
+                comparison > 0 -> current
+                comparison == 0 -> listOf(current, next).minBy { it.path.size }
+                else -> actualNext
+            }
+        }
+        result.copy(local = initial, path = result.path + localId)
```

Comment:
Why a copy instead of a new object creation? usually, I try to avoid using `copy` method on `data class`es